resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/BulkheadAspect.java

```diff
@@ -61,7 +62,9 @@ public Object bulkheadAroundAdvice(ProceedingJoinPoint proceedingJoinPoint, Bulk
 		}
 		String backend = backendMonitored.name();
 		io.github.resilience4j.bulkhead.Bulkhead bulkhead = getOrCreateBulkhead(methodName, backend);
-		return handleJoinPoint(proceedingJoinPoint, bulkhead, methodName);
+		RecoveryFunction recovery = backendMonitored.recovery().newInstance();
```

Comment:
Maybe we can create an instance of the RecoveryFunction only when needed. What do you think?

Reply:
I agree with @RobWin  , let us make on need basis and plus can u please update the documentation as well for the same ?